src/core/Akka.Streams/Implementation/Fusing/StreamOfStreams.cs

```diff
@@ -419,9 +419,9 @@ public void OnPush()
                     }
                     else
                     {
-                        if (_activeSubstreams.Count + _closedSubstreams.Count == _stage._maxSubstreams)
+                        if (_stage._maxSubstreams > -1 && _activeSubstreams.Count + _closedSubstreams.Count == _stage._maxSubstreams)
```

Comment:
Actual fix. Only check substream overflow condition if `_maxSubstream` is greater than or equals to 0.